<robot name="non_holonomy_robot">
 
 <link name="world"/>
 <link name="t_x_link"/>
 <link name="t_z_link"/>
 
 <link name="base_link">
    <inertial>
     <origin xyz="0 0 0.0" rpy="0 0 0"/>
     <mass value="1"/>
     <inertia ixx="0.00832"  ixy="0"  ixz="0" iyy="0.0908" iyz="0" izz="0.084" />
   </inertial>

   <visual>
     <origin xyz="0 0 0" rpy="0 0 0" />
     <geometry>
       <box size="1 0.1 0.3" />
     </geometry>
     <material name="Cyan">
       <color rgba="0 1.0 1.0 1.0"/>
     </material>
   </visual>

   <collision>
     <origin xyz="0 0 0" rpy="0 0 0"/>
     <geometry>
       <box size="1 0.1 0.3" />
     </geometry>
   </collision>
 </link>
    
 <link name="left_link">
    <inertial>
     <origin xyz="-0.5 0 0.0" rpy="0 0 0"/>
     <mass value="1"/>
     <inertia ixx="0.00832"  ixy="0"  ixz="0" iyy="0.0908" iyz="0" izz="0.084" />
   </inertial>

   <visual>
     <origin xyz="-0.5 0 0" rpy="0 0 0" />
     <geometry>
       <box size="1 0.1 0.3" />
     </geometry>
     <material name="Yellow">
       <color rgba="1.0 0.3 0.3 1.0"/>
     </material>
   </visual>

   <collision>
     <origin xyz="-0.5 0 0" rpy="0 0 0"/>
     <geometry>
       <box size="1 0.1 0.3" />
     </geometry>
   </collision>
 </link>
   
 <link name="right_link">
    <inertial>
     <origin xyz="0.5 0 0.0" rpy="0 0 0"/>
     <mass value="1"/>
     <inertia ixx="0.00832"  ixy="0"  ixz="0" iyy="0.0908" iyz="0" izz="0.084" />
   </inertial>

   <visual>
     <origin xyz="0.5 0 0" rpy="0 0 0" />
     <geometry>
       <box size="1 0.1 0.3" />
     </geometry>
     <material name="Blue">
       <color rgba="0.3 0.3 1.0 1.0"/>
     </material>
   </visual>

   <collision>
     <origin xyz="0.5 0 0" rpy="0 0 0"/>
     <geometry>
       <box size="1 0.1 0.3" />
     </geometry>
   </collision>
 </link>

 <joint name="t_x_joint" type="prismatic">
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="1 0 0"/>
    <parent link="world"/>
    <child link="t_x_link"/>
    <limit effort="300" velocity="10.0" lower="-10.0" upper="10.0"/>
 </joint>
 
 <joint name="t_z_joint" type="prismatic">
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="0 0 1"/>
    <parent link="t_x_link"/>
    <child link="t_z_link"/>
    <limit effort="300" velocity="10.0" lower="-10.0" upper="10.0"/>
 </joint>
 
 <joint name="rot_joint" type="revolute">
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="0 -1 0"/>
    <parent link="t_z_link"/>
    <child link="base_link"/>
    <limit effort="300" velocity="10.0" lower="-100.0" upper="100.0"/>
 </joint>
 
 <joint name="left_joint" type="revolute">
    <origin xyz="-0.5 0 0" rpy="0 0 0"/>
    <axis xyz="0 -1 0"/>
    <parent link="base_link"/>
    <child link="left_link"/>
    <limit effort="300" velocity="10.0" lower="-2" upper="2"/>
 </joint>
 
 <joint name="right_joint" type="revolute">
    <origin xyz="0.5 0 0" rpy="0 0 0"/>
    <axis xyz="0 -1 0"/>
    <parent link="base_link"/>
    <child link="right_link"/>
    <limit effort="300" velocity="10.0" lower="-2" upper="2"/>
 </joint>
 
</robot>
